### DISCUSSION TITLE
milvus对接grafana，关于后台查询prometheus一些语句不够全的问题

### DISCUSSION THREAD

**Discussion by xkz130** [milvus-io/milvus#22319] [Q&A and General discussion]
调试grfana的监控页面时，发现prometheus的后台查询只能查到协调器coord相关的服务监控图标，而node相关服务的都没有，导致监控只能显示coord协调器服务的监控，node相关的服务数据时空的

![image](https://user-images.githubusercontent.com/46994912/220270136-bbf165b7-205d-4286-a431-bb074961c66c.png)

![image](https://user-images.githubusercontent.com/46994912/220270719-fc6d7e71-27b9-460f-882c-826797dc89fa.png)

![image](https://user-images.githubusercontent.com/46994912/220270818-9e2ed2a9-0946-4621-beb6-1deeec502b95.png)

**Comment by LoveEachDay** [comment]
@xkz130 Could you double check whether  prometheus scrape targets includes milvus nodes?

**Reply by xkz130** [reply]
我显示的是所有节点的，不是这个问题，比如说图中node监控里Collection Loaded Num的监控语句用的是 milvus_querynode_collection_num
 在prometheus中搜不到这个milvus_querynode_collection_num
![image](https://user-images.githubusercontent.com/46994912/220306043-afc96796-97ee-4ef2-8a7b-637c88ab4d17.png)
 而能够正常显示的是可以搜出来的 比如说coord服务中的 milvus_rootcoord_proxy_num
![image](https://user-images.githubusercontent.com/46994912/220306278-a8bd2937-a505-4f12-b4b7-5e88a36f58c4.png)

**Reply by LoveEachDay** [reply]
I know your problem. You need verify whether prometheus can scrape the metrics from querynode for example. 
You can verify from the prometheus web ui under `Status` -> `Target` menu.
